apps/src/gamelab/AnimationPicker/AnimationPreview.jsx

```diff
@@ -56,6 +58,16 @@ const AnimationPreview = React.createClass({
   },
 
   advanceFrame: function () {
+    // If the animation shouldn't loop, include a 2.5 second timeout after playing.
+    if (this.props.animationProps.looping === false && (this.state.currentFrame % this.props.animationProps.frameCount) === 0) {
+      this.setState({
+        currentFrame: (this.state.currentFrame + 1) % this.props.animationProps.frameCount
+      });
+      clearTimeout(this.timeout_);
+      this.timeout_ = setTimeout(this.advanceFrame, 2500);
+      return;
```

Comment:
The early return here falls right in the middle of the method; it doesn't feel like an early-out because some argument was no good.  I'd prefer if/else in a situation like this.

That said, the two cases are really similar; in fact, I think the only difference between your two cases is the length of the timeout you set.  How about something vaguely like this instead?
```javascript
const animationIsOver = !looping && currentFrame % frameCount === 0;
const timeoutDuration = animationIsOver ? 2500 : 33 * frameDelay;
this.timeout_ = setTimeout(this.advanceFrame, timeoutDuration);
```